Assets/LeapMotionModules/UI/Interaction/InteractionManager.cs

```diff
@@ -0,0 +1,293 @@
+      _interactionHands[1] = new InteractionHand(this, () => { return Hands.FixedRight; }, Chirality.Right, WorldHoverActivationRadius, WorldTouchActivationRadius);
+
+      if (_autoGenerateLayers) {
+        AutoGenerateLayers();
+        AutoSetupCollisionLayers();
+      }
+    }
+
+    void Start() {
+      if (Provider == null) {
+        Debug.LogError("[InteractionManager] No LeapServiceProvider found.");
+      }
+    }
+
+    void FixedUpdate() {
+      OnPrePhysicalUpdate();
+
+      foreach (var interactionHand in _interactionHands) {
+        interactionHand.FixedUpdateHand(enableHovering, enableContact, enableGrasping);
+      }
+
+      // Apply soft contacts from both hands in unified solve.
+      // (This will clear softContacts and originalVelocities as well.)
+      if (_softContacts.Count > 0) {
+        PhysicsUtility.ApplySoftContacts(_softContacts, _softContactOriginalVelocities);
+      }
+
+      foreach (var interactionObj in _interactionBehaviours) {
+        interactionObj.FixedUpdateObject();
+      }
+
+      OnPostPhysicalUpdate();
+    }
+
+    void Update() {
+      if (Provider != null) {
+        _providerScale = Provider.transform.lossyScale.x;
+
+        foreach (var interactionHand in _interactionHands) {
+          interactionHand.HoverActivationRadius = WorldHoverActivationRadius;
+          interactionHand.TouchActivationRadius = WorldTouchActivationRadius;
+        }
+      }
+    }
+
+    void LateUpdate() {
+      OnGraphicalUpdate();
+    }
+
+    #region Object Registration
+
+
+    public void RegisterInteractionBehaviour(InteractionBehaviourBase interactionObj) {
+      _interactionBehaviours.Add(interactionObj);
+      _rigidbodyRegistry[interactionObj.Rigidbody] = interactionObj;
+    }
+
+    #endregion
+
+    /// <summary>
+    /// Returns the InteractionHand object that corresponds to the given Hand object.
+    /// 
+    /// Currently, the InteractionManager supports only two InteractionHands at one time: one player's left and right hands.
+    /// </summary>
+    public InteractionHand GetInteractionHand(Hand hand) {
+      if (hand.Handedness() == Chirality.Left) {
+        return _interactionHands[0];
+      }
+      else {
+        return _interactionHands[1];
+      }
+    }
+
+    /// <summary> Returns true if the object was released from a grasped hand, or false if the object was not held in the first place. </summary>
+    public bool ReleaseObjectFromGrasp(InteractionBehaviourBase interactionObj) {
+      if (!_interactionBehaviours.Contains(interactionObj)) {
+        Debug.LogError("ReleaseObjectFromGrasp was called, but the interaction object " + interactionObj.transform.name + " is not registered"
+          + " with this InteractionManager.");
+        return false;
+      }
+
+      foreach (var hand in _interactionHands) {
+        if (hand.IsGrasping(interactionObj)) {
+          hand.ReleaseGrasp();
+          return true;
```

Comment:
For cases where more than one hand is holding onto an object, you don't want to bail out at the first hand found.